### PR TITLE
[SPARK-19937] Collect metrics for remote bytes read to disk during shuffle.

### PR DESCRIPTION
**Pull request by jinxing64**
In current code(https://github.com/apache/spark/pull/16989), big blocks are shuffled to disk. 
This pr proposes to collect metrics for remote bytes fetched to disk.

### CODE REVIEW

File: project/MimaExcludes.scala [hunk 1]
```diff
@@ -38,6 +38,10 @@ object MimaExcludes {
   lazy val v23excludes = v22excludes ++ Seq(
     // [SPARK-20495][SQL] Add StorageLevel to cacheTable API
     ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.sql.catalog.Catalog.cacheTable")
+  ) ++ Seq(
```
Comment (vanzin) [inline project/MimaExcludes.scala]: Could you follow the same style as `v22excludes`, which avoids this extra line?

**Review by vanzin — COMMENTED** [review]
Looks ok. At first I wasn't really sold on the name but can't think of something better either.

**Comment by jinxing64** [comment]
@vanzin
Would you mind give more comments when have time ? And I can continue working on this :)

**Comment by vanzin** [comment]
I was just waiting in case anyone else wanted to comment.

LGTM, merging to master.

**Pull request closed without merging**